presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -219,6 +225,31 @@ public static Slice stGeomFromBinary(@SqlType(StandardTypes.VARBINARY) Slice inp
         return serialize(geomFromBinary(input));
     }
 
+    @Description("Converts a Geometry object to a SphericalGeography object")
+    @ScalarFunction("to_spherical_geography")
+    @SqlType(SPHERICAL_GEOGRAPHY_TYPE_NAME)
+    public static Slice toSphericalGeography(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
+    {
+        // "every point in input is in range" <=> "the envelope of input is in range"
+        Envelope envelope = deserializeEnvelope(input);
+        if (envelope != null && !envelope.isEmpty()) {
```

Comment:
Is `envelope.isEmpty()` possible here?

Reply:
looks like not - deserializeEnvelope() always returns null for empty shapes and other call sites like stXMax() only does null check on the envelope.
I'll remove the envelope.isEmpty() check.